git-extra/git-sdk.sh

```diff
@@ -87,6 +87,10 @@ sdk () {
 		printf "%s " build-extra git git-extra MINGW-packages \
 			MSYS2-packages msys2-runtime
 		;;
+	valid_build_targets)
+		printf "%s " git-and-installer $(sdk valid_projects | tr ' ' '\n' |
+			grep -v '^\(build-extra\|\(MINGW\|MSYS2\)-packages\)')
```

Comment:
Maybe use `grep -E` here?

Reply:
I try to avoid non-standard options; it is a bad habit to get into when you have to contribute patches to Git's test suite (where we really need portability)...

Reply:
But... would a `grep` that doesn't understand `-E` grok the extended syntax, even if the metacharacters are quoted? 🤔

Reply:
I think so: https://www.regular-expressions.info/posix.html#bre

Reply:
I think so: https://www.regular-expressions.info/posix.html#bre